### PR TITLE
Speed-up CI end-to-end tests

### PR DESCRIPTION
**Pull request by adizere**
Close: #582

______

For contributor use:

- [X] Updated the __Unreleased__ section of [CHANGELOG.md](https://github.com/informalsystems/ibc-rs/blob/master/CHANGELOG.md) with the issue.
- [X] If applicable: Unit tests written, added test to CI.
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -2,13 +2,22 @@ name: End to End testing
...
     steps:
       - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --workspace --all-targets
```
Comment (adizere) [inline .github/workflows/e2e.yaml:16]: The binary is built in debug mode (without `--release`). This is what accounts primarily for shortening the workflow cycle.

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -44,10 +44,6 @@ jobs:
         with:
           toolchain: stable
           override: true
-      - uses: actions-rs/cargo@v1
-        with:
-          command: build
```
Comment (adizere) [inline .github/workflows/rust.yml:49]: This build was not necessary.

File: ci/relayer.Dockerfile [hunk 3]
```diff
@@ -35,11 +6,11 @@ LABEL maintainer="hello@informal.systems"
 
 ARG RELEASE
 
-# Add jq and Python 3
-RUN apt-get update -y && apt-get install python3 jq -y
```
Comment (adizere) [inline ci/relayer.Dockerfile:39]: `jq` wasn't used in the relayer image.

File: .github/workflows/e2e.yaml (hunk 1 above)
Comment (romac) [inline .github/workflows/e2e.yaml:16]: While we are at it we can also remove `--all-targets` as we don't not need to build tests, etc.

```suggestion
          args: --workspace
```

**Review by andynog — APPROVED** [review]
The idea of building the relayer inside the Docker container was to avoid any side effects of building in one platform but since the Docker container is Debian and the CI uses Ubuntu, I think it's all good and the speed improvement are great

**Comment by adizere** [comment]
> The idea of building the relayer inside the Docker container was to avoid any side effects of building in one platform but since the Docker container is Debian and the CI uses Ubuntu, I think it's all good and the speed improvement are great

Hmm this is something I overlooked!

In a future PR, maybe we run the relayer binary directly from Github actions? This would further simplify the workflow (and probably reduce the time).

**Pull request merged**